test/functional/api/window_spec.lua

```diff
@@ -347,4 +347,32 @@ describe('API/win', function()
       eq('', funcs.getcmdwintype())
     end)
   end)
+
+  describe('hide', function()
+    it('can hide current window', function()
+      local oldwin = meths.get_current_win()
+      command('split')
+      local newwin = meths.get_current_win()
+      meths.win_hide(newwin)
+      eq({oldwin}, meths.list_wins())
+    end)
+    it('can hide noncurrent window', function()
+      local oldwin = meths.get_current_win()
+      command('split')
+      local newwin = meths.get_current_win()
+      meths.win_hide(oldwin)
+      eq({newwin}, meths.list_wins())
+    end)
+    it('does not close the buffer', function()
+      local oldwin = meths.get_current_win()
+      local oldbuf = meths.get_current_buf()
+      local buf = meths.create_buf(true, false)
+      local newwin = meths.open_win(buf, true, {
+        relative='win', row=3, col=3, width=12, height=3
+      })
+      meths.win_hide(newwin)
+      eq({oldwin}, meths.list_wins())
+      eq({oldbuf, buf}, meths.list_bufs())
+    end)
```

Comment:
can you add a test where you set `bufhidden` for the buffer to be `wipe` and then make sure it wipes it when you hide the window? I think that's the expected behavior.